File: app/src/main/java/com/nfcalarmclock/alarm/options/maxsnooze/NacMaxSnoozeDialog.java
package com.nfcalarmclock.alarm.options.maxsnooze;

import android.app.AlertDialog;
import android.app.Dialog;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.nfcalarmclock.view.dialog.NacScrollablePickerDialogFragment;
import com.nfcalarmclock.shared.NacSharedConstants;
import com.nfcalarmclock.R;

import java.util.List;

/**
 * Select the max number of snoozes allowed for an alarm.
 */
public class NacMaxSnoozeDialog
	extends NacScrollablePickerDialogFragment
{

	/**
	 * Tag for the class.
	 */
	public static final String TAG = "NacMaxSnoozeDialog";

	/**
	 * Get the list of values for the scrollable picker.
	 *
	 * @return The list of values for the scrollable picker for the scrollable
	 *     picker.
	 */
	public List<String> getScrollablePickerValues()
	{
		NacSharedConstants cons = this.getSharedConstants();
		return cons.getMaxSnoozeSummaries();
	}

	/**
	 */
	@NonNull
	@Override
	public Dialog onCreateDialog(@Nullable Bundle savedInstanceState)
	{
		this.setupSharedPreferences();

		NacSharedConstants cons = this.getSharedConstants();

		return new AlertDialog.Builder(requireContext())
			.setTitle(cons.getMaxSnooze())
			.setPositiveButton(cons.getActionOk(), (dialog, which) ->
				this.callOnScrollablePickerOptionSelectedListener())
			.setNegativeButton(cons.getActionCancel(), (dialog, which) -> {})
			.setView(R.layout.dlg_scrollable_picker)
			.create();
	}

}

File: app/src/main/java/com/nfcalarmclock/maxsnooze/NacMaxSnoozeDialog.kt
package com.nfcalarmclock.maxsnooze

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import com.nfcalarmclock.R
import com.nfcalarmclock.view.dialog.NacScrollablePickerDialogFragment

/**
 * Select the max number of snoozes allowed for an alarm.
 */
class NacMaxSnoozeDialog
	: NacScrollablePickerDialogFragment()
{

	companion object
	{

		/**
		 * Tag for the class.
		 */
		const val TAG = "NacMaxSnoozeDialog"

	}

	/**
	 * The list of values for the scrollable picker.
	 */
	override val scrollablePickerValues: List<String>
		get()
		{
			val array = resources.getStringArray(R.array.max_snooze_summaries)

			return array.asList()
		}

	/**
	 * Called when the dialog is created.
	 */
	override fun onCreateDialog(savedInstanceState: Bundle?): Dialog
	{
		// Setup the shared preferences
		setupSharedPreferences()

		// Create the dialog
		return AlertDialog.Builder(requireContext())
			.setTitle(R.string.max_snooze)
			.setPositiveButton(R.string.action_ok) { _, _ ->

				// Call the listener
				callOnScrollablePickerOptionSelectedListener()

			}
			.setNegativeButton(R.string.action_cancel, null)
			.setView(R.layout.dlg_scrollable_picker)
			.create()
	}

}